docs/designers-developers/developers/backward-compatibility/README.md

```diff
@@ -1 +1,65 @@
+
+* should be constrained as much as possible to a small surface area of the API.
+* should be documented as clearly as possible to third-party developers using dev notes.
+
+## What qualifies as a production public API
+
+The Gutenberg code base is composed of two different types of packages: 
+ - **production packages**: these are packages that are shipped as WordPress scripts (example: wp-components, wp-editor...).
+ - **development packages**: these are made out of developer tools that can be used by third-party developers to lint, test, format and build their themes and plugins (example: @wordpress/scrips, @wordpress/env...). Typically, these are consumed as npm dependencies in third-party projects.
+
+Backward compatibility guarantee only applies to the production packages as updates happen through WordPress upgrades.
+ 
+Production packages use the `wp` global variable to provide APIs to third-party developers. These APIs can be JavaScript functions, variables and React components.
+
+### How to preserve backward compatibility for a JavaScript function
+
+* The name of the function should not change.
+* The order of the arguments of the function should not change.
+* The function's returned value type should not change.
+* Changes to arguments (new arguments, modification of semantics) is possible if we guarantee that all previous calls are still possible.
+
+### How to preserve backward compatibility for a React Component
+
+* The name of the component should not change.
+* The props of the components should not be removed.
+* Existing prop values should continue to be supported. If a component accepts a function as prop, we can update the component to accept a new type for the same prop, but it shouldn't break existing usage.
+* Adding new props is allowed.
+* React Context dependencies can only be added or removed if we ensure the previous context contract is not breaking.
+
+### How to preserve backward compatibility for a Block
+
+* Existing usage of the block should not break or be marked as invalid when the editor is loaded.
+* The styling of the existing blocks should be guaranteed.
+* Markup changes should be limited to the minimum but If a block need to change its saved markup making previous versions invalid, a [**deprecated version**](/docs/designers-developers/developers/block-api/block-deprecation.md) for the block should be added.
+
+## Class names and DOM updates
+
+Class names and DOM nodes used inside the tree of React components are not considered part of the public API and can be modified. 
+
+Changes to these should be done with caution as it can affect the styling and behavior of third-party code (Even if they should not rely use these in first place). Keep the old ones if possible. If not, document these changes and write a dev note about the changes.
```

Comment:
```suggestion
Changes to these should be done with caution as it can affect the styling and behavior of third-party code (Even if they should not rely on these in the first place). Keep the old ones if possible. If not, document the changes and write a Dev Note.
```

Reply:
Can you lower case dev note :)